### DISCUSSION TITLE
Is it possible to have all path separators return ‘/’ on Windows?

### DISCUSSION THREAD

**Discussion by lbgitjp** [dotnet/runtime#104207] [Q&A] [answered]
FileSystemInfo‘s Properties such as 'FullPath' and 'FullName', on Windows platform, it will use '\\' as the path separator,  and on other platforms will use '/' as the separator.  I need to convert these paths to URLs for web, so when i get FullName on windows, i don't want it return "c:\\webroot\\dir1\\test.txt", i want it return "c:/webroot/dir1/test.txt", is it possible?

Thanks!

**Comment by MichalPetryka** [comment]
Why not use the [URI constructor](https://learn.microsoft.com/en-us/dotnet/api/system.uri.-ctor?view=net-8.0#system-uri-ctor(system-string))?

**Reply by lbgitjp** [reply]
I need show all files in folder "c:\webroot\dir1\" to web, so i need convert every file's fullname to url, 
e.g  i need convert "c:\webroot\dir1\test.gif" to "/dir1/test.gif", so how to do this by use [URI constructor](https://learn.microsoft.com/en-us/dotnet/api/system.uri.-ctor?view=net-8.0#system-uri-ctor(system-string))?

Thanks!

**Comment by andriivitiv** [comment, marked as answer]
```cs
var rootUri = new Uri(@"c:\webroot\");
var fullUri = new Uri(@"c:\webroot\dir1\test.gif");
var relativeUri = rootUri.MakeRelativeUri(fullUri);

Console.WriteLine(relativeUri); // => dir1/test.gif
```

**Reply by lbgitjp** [reply]
I didn't know could pass an operating system path to a URI until you told me.

Thanks!
